### DISCUSSION TITLE
"Everything" in one .xll?

### DISCUSSION THREAD

**Discussion by dutch-rob** [cunnane/xloil#34] [Ideas]
I have been using xloil and it is helping me a lot. So much, that I want to share some things I developed with someone else. However,  they don't already have Python installed, therefore would need to do that, install the necessary Python packages, etc. without knowing Python.
That made me wonder: Would it be difficult to make xloil able to produce a single .xll file that includes everything needed to run well in Excel?
It is possible to make single file executables with pyinstaller or nuitka, so (probably because I know way too little about how those work) it seems something like that may work for making .xll's with xloil as well.

**Comment by cunnane** [comment]
@dutch-rob you could zip up a python distro (I'd suggest not using anaconda here as it's massive), and unpack it on the user's machine - python doesn't need any more installing than that.  The user could run xlOil but dropping the xll in their session (xlOil will look in the xll's directory for an ini file, which would need to have the paths pointing to the right place).   Alternatively:
* xlOil.ini should be copied to %APPDATA%\xlOil` 
* xlOil.xll should be copied to the `XLSTART` folder.  
* The paths in xlOil.ini need to point to the python distribution and xlOil binaries

There's [a powershell script you could adapt](https://github.com/cunnane/xloil/blob/master/tools/xlOil_Install.ps1) which even edits the paths in the *xloil.ini* file.  It hasn't been used in a while though, so may need a refresh.  If powershell isn't your game, I could probably get this approach running fairly easily.

I've looked at *pyinstaller* and I'm sure with some delicate changes it could be coaxed into producing an XLL which unpack a python distribution, just like their exes do.   This would certainly be the nicer solution although it will me some time though as I'll need to carefully go through their C bootstrapper code.

**Comment by dutch-rob** [comment]
Thanks for the advice, I will try that! 
Rob
